rpc/legacyrpc/methods.go

```diff
@@ -3202,13 +3203,17 @@ func signMessage(icmd interface{}, w *wallet.Wallet) (interface{}, error) {
 	wire.WriteVarString(&buf, 0, "Decred Signed Message:\n")
 	wire.WriteVarString(&buf, 0, cmd.Message)
 	messageHash := chainhash.HashFuncB(buf.Bytes())
-	r, s, err := chainec.Secp256k1.Sign(privKey, messageHash)
+	pkCast, ok := privKey.(*secp256k1.PrivateKey)
+	if !ok {
+		return nil, fmt.Errorf("Unable to create secp256k1.PrivateKey" +
```

Comment:
I wouldn't use actual type names here since this is possible to hit as a user by using a non-secp256k1 P2PKH address.  Instead, I would say something such as "The address must correspond to a secp256k1 public key", and perhaps even include in the error what type of DSA and/or curve the address actually represents.